tapiriik/services/Aerobia/aerobia.py

```diff
@@ -0,0 +1,340 @@
+        60 : ActivityType.Other, #FISHING
+        67 : ActivityType.Other, #SCOOTER
+        15 : ActivityType.Other, #WINDSURFING
+        42 : ActivityType.Other, #SQUASH
+        41 : ActivityType.Other, #SKATEBOARD
+        75 : ActivityType.Other, #STEPPER
+        29 : ActivityType.Other, #DANCING
+        40 : ActivityType.Other, #TENNIS
+        37 : ActivityType.Other, #TABLE TENNIS
+        81 : ActivityType.Other, #OUTDOOR FITNESS
+        31 : ActivityType.Other, #FOOTBALL
+        59 : ActivityType.Other, #FENCING
+        39 : ActivityType.Other, #FIGURE SKATING
+        34 : ActivityType.Other, #HOCKEY
+        82 : ActivityType.Other, #CHESS
+
+        68 : ActivityType.Other
+    }
+
+    SupportedActivities = list(_activityMappings.keys())
+
+    SupportsHR = SupportsCadence = True
+
+    SupportsActivityDeletion = True
+
+    _urlRoot = "http://aerobia.ru/"
+    _apiRoot = "http://aerobia.ru/api/"
+    _loginUrlRoot = _apiRoot + "sign_in"
+    _workoutsUrl = _apiRoot + "workouts"
+    _workoutUrl = _apiRoot + "workouts/{id}.json"
+    _uploadsUrl = _apiRoot + "uploads.json"
+
+    def _patch_user_agent(self):
+        from tapiriik.requests_lib import patch_requests_user_agent
+        # Without user-agent patch aerobia requests doesn't work
+        patch_requests_user_agent('Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; rv:11.0) like Gecko')
+
+    def _get_auth_data(self, record=None, username=None, password=None):
+        from tapiriik.auth.credential_storage import CredentialStore
+
+        if record:
+            #  longing for C style overloads...
+            password = CredentialStore.Decrypt(record.ExtendedAuthorization["Password"])
+            username = CredentialStore.Decrypt(record.ExtendedAuthorization["Email"])
+
+        self._patch_user_agent()
+        request_parameters = {"user[email]": username, "user[password]": password}
+        res = requests.post(self._loginUrlRoot, data=request_parameters)
+
+        if res.status_code != 200:
+            raise APIException("Login exception {} - {}".format(res.status_code, res.text), user_exception=UserException(UserExceptionType.Authorization))
+
+        res_xml = etree.fromstring(res.text.encode('utf-8'))
+
+        info = res_xml.find("info")
+        if info.get("status") != "ok":
+            raise APIException(info.get("description"), user_exception=UserException(UserExceptionType.Authorization))
+
+        user_id = int(res_xml.find("user/id").get("value"))
+        user_token = res_xml.find("user/authentication_token").get("value")
+
+        return user_id, user_token
+
+    def _call(self, serviceRecord, request_call, *args):
+        retry_count = 3
+        resp = None
+        for i in range(0, retry_count):
+            try:
+                resp = request_call(args)
+                break
+            except APIException:
+                # try to refresh token first
+                self._refresh_token(serviceRecord)
```

Comment:
How long do the tokens last? Rather than store one with the service record and attempt to use it later, it might make more sense to start every session with a fresh token if we know the old one is almost always going to be expired by the time we try to use it.